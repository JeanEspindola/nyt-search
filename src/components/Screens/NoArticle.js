import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import WithLocale from '../HOC/WithLocale';

const NoArticle = ({ locale }) => (
  <Fragment>
    <p>{locale.no_details}</p>
  </Fragment>
);

NoArticle.propTypes = {
  locale: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default WithLocale(NoArticle);
