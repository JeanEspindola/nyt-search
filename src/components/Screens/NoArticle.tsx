import { FormattedMessage } from 'react-intl'

const NoArticle = () => (
  <p>
    <FormattedMessage id={'noDetails'} />
  </p>
);

export default NoArticle;
