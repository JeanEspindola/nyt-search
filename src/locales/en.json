{
  "go_to_results": "Go to Results Page",
  "header_title": "\"The New York Times\" article search application",
  "next_page": "Next Page",
  "no_details": "No details to show.",
  "no_results": "There are no results to display.",
  "previous_page": "Previous Page",
  "read_article": "Read the Full Article",
  "results": "Results:",
  "search_terms": "Search your terms here.",
  "type_query": "Type search query term in here:"
}