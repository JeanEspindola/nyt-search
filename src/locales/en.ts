// eslint-disable-next-line import/no-anonymous-default-export
export default {
  goToResults: "Go to Results Page",
  headerTitle: "\"The New York Times\" article search application",
  nextPage: "Next Page",
  noDetails: "No details to show.",
  noResults: "There are no results to display.",
  previousPage: "Previous Page",
  readArticle: "Read the Full Article",
  results: "Results:",
  searchTerms: "Search your terms here.",
  typeQuery: "Type search query term in here:"
}
