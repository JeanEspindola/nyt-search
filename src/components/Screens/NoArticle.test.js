import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NoArticle from './NoArticle';
import en from '../../locales/en.json';

describe('EmptyList', () => {
  let Component;

  beforeEach(() => {
    Component = shallow(
      <NoArticle />,
      { context: { locale: en } },
    ).find('NoArticle')
      .shallow();
  });

  it('renders without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });
});
