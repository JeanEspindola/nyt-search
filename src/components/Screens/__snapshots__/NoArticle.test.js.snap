// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList renders without crashing 1`] = `
<React.Fragment>
  <p>
    No details to show.
  </p>
</React.Fragment>
`;
